app/commands/user/premium/update.rb

```diff
@@ -0,0 +1,43 @@
+class User::Premium::Update
+  include Mandate
+
+  initialize_with :user
+
+  def call
+    return if user.premium_until == premium_until
+
+    if expire?
+      User::Premium::Expire.(user)
+    else
```

Comment:
Should this be:
```suggestion
    elsif user.premium_until.nil?
```

Reply:
No, because of the "sliding window" aspect of it. If a new payment was made, the `premium_until` time needs to be updated. And the preceding `return if user.premium_until == premium_until` prevents unnecessary updating.

Reply:
But I don't understand why we call `Join` a second time in that case. Shouldn't we be calling something to update `premium_until` instead?

Reply:
That's basically what `Join` does